ddl/ddl_api.go

```diff
@@ -323,7 +323,7 @@ func columnDefToCol(ctx sessionctx.Context, offset int, colDef *ast.ColumnDef) (
 				removeOnUpdateNowFlag(col)
 			case ast.ColumnOptionOnUpdate:
 				// TODO: Support other time functions.
-				if !expression.IsCurrentTimestampExpr(v.Expr) {
+				if expression.IsCurrentTimestampExpr(v.Expr) {
```

Comment:
Why change this check?

Reply:
This code was not tested before, logic is the opposite.